frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1282,7 +1282,15 @@ dbg:guard(ReaderHighlight, "lookup",
     end)
 
 function ReaderHighlight:getSelectedWordContext(nb_words)
-    if not self.ui.rolling or not self.selected_text then return nil end
+    if not self.selected_text then return end
+    if not self.ui.rolling then
+        local ok, prev_context, next_context = pcall(self.ui.document.getSelectedWordContext, self.ui.document, self.selected_text.text, self.selected_text.pos0, nb_words)
+        if ok then
+            return prev_context, next_context
+        else
+            return nil
+        end
+    end
     local pos_start = self.selected_text.pos0
```

Comment:
I'm thinking about lowering the crengine part into `credocument.lua` and just call `sel.ui.document.getSelectedWordContext' here.

Reply:
No strong opinion.
You could, for balance with the non-cre stuff.
But I don't mind seeing it all here as there's already many bits where we see how it works with xpointers.
Also, 2 things if you go with putting it in credocument:
- there's not many pcall() in credocument (only one, when we create a document, so we don't crash if libcrengine is badly compiled), so from a arch point of view, you should try to not need any pcall (and keep one here if needed when you call the new credocument method).
- think about whether you want it cached or not by the crengine method calls result cache

Reply:
>there's not many pcall() in credocument

Removed pcalls as would be expected

>think about whether you want it cached or not 

Didn't really find what to cache so stick to previous logic...